### PR TITLE
Added SFP sensor discovery for Procurve devices

### PR DESCRIPTION
**Pull request by angryp**
Procurve devices have SPF ports. If those are filled with SPF modules with DOM support, sensor data will be available through SNMP. Tested on J9775A 2530-48G switches. 
Please bear with my coding. I'm far from programming and adjusted discoveries from Comware as a sample. Let me know if something should be adjusted.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/sensors/current/procurve.inc.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        $limit_low                 = $entry['hpicfXcvrBiasLoAlarm'] / $divisor_alarm;
+        $warn_limit_low            = $entry['hpicfXcvrBiasLoWarn'] / $divisor_alarm;
+        $limit                     = $entry['hpicfXcvrBiasHiAlarm'] / $divisor_alarm;
+        $warn_limit                = $entry['hpicfXcvrBiasHiWarn'] / $divisor_alarm;
+        $current                   = $entry['hpicfXcvrBias'] / $divisor;
+        $entPhysicalIndex          = $index;
+        $entPhysicalIndex_measured = 'ports';
+        foreach ($dbquery as $dbindex => $dbresult) {
+            $descr = makeshortif($dbresult['ifDescr']) . ' Port Bias Current';
+            discover_sensor($valid['sensor'], 'current', $device, $oid, 'bias-' . $index, 'procurve', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
```
Comment (laf) [inline includes/discovery/sensors/current/procurve.inc.php]: Change `'bias-' . $index` for `"hpicfXcvrBias.$index"`

File: includes/discovery/sensors/dbm/procurve.inc.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        $limit_low                 = round(uw_to_dbm($entry['hpicfXcvrRcvPwrLoAlarm'] / 10), 2);
+        $warn_limit_low            = round(uw_to_dbm($entry['hpicfXcvrRcvPwrLoWarn'] / 10), 2);
+        $limit                     = round(uw_to_dbm($entry['hpicfXcvrRcvPwrHiAlarm'] / 10), 2);
+        $warn_limit                = round(uw_to_dbm($entry['hpicfXcvrRcvPwrHiWarn'] / 10), 2);
+        $current                   = $entry['hpicfXcvrRxPower'] / $divisor;
+        $entPhysicalIndex          = $index;
+        $entPhysicalIndex_measured = 'ports';
+        foreach ($dbquery as $dbindex => $dbresult) {
+            $descr = makeshortif($dbresult['ifDescr']) . ' Port Receive Power';
+            discover_sensor($valid['sensor'], 'dbm', $device, $oid, 'rx-' . $index, 'procurve', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
```
Comment (laf) [inline includes/discovery/sensors/dbm/procurve.inc.php]: Change `'rx-' . $index` for `"hpicfXcvrRxPower.$index"`

File: includes/discovery/sensors/dbm/procurve.inc.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        $limit_low                 = round(uw_to_dbm($entry['hpicfXcvrPwrOutLoAlarm'] / 10), 2);
+        $warn_limit_low            = round(uw_to_dbm($entry['hpicfXcvrPwrOutLoWarn'] / 10), 2);
+        $limit                     = round(uw_to_dbm($entry['hpicfXcvrPwrOutHiAlarm'] / 10), 2);
+        $warn_limit                = round(uw_to_dbm($entry['hpicfXcvrPwrOutHiWarn'] / 10), 2);
+        $current                   = $entry['hpicfXcvrTxPower'] / $divisor;
+        $entPhysicalIndex          = $index;
+        $entPhysicalIndex_measured = 'ports';
+        foreach ($dbquery as $dbindex => $dbresult) {
+            $descr = makeshortif($dbresult['ifDescr']) . ' Port Transmit Power';
+            discover_sensor($valid['sensor'], 'dbm', $device, $oid, 'tx-' . $index, 'procurve', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
```
Comment (laf) [inline includes/discovery/sensors/dbm/procurve.inc.php]: Change `'tx-' . $index` for `"hpicfXcvrTxPower.$index"`

File: includes/discovery/sensors/temperature/procurve.inc.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    foreach ($entphydata as $index) {
+        foreach ($tempdata as $tempindex => $value) {
+            if ($index['entPhysicalIndex'] == $tempindex && $value['hpicfXcvrTemp'] != 0) {
+                $cur_oid = '.1.3.6.1.4.1.11.2.14.11.5.1.82.1.1.1.';
+                discover_sensor(
+                    $valid['sensor'],
+                    'temperature',
+                    $device,
+                    $cur_oid . $tempindex,
+                    'temp-' . $tempindex,
```
Comment (laf) [inline includes/discovery/sensors/temperature/procurve.inc.php]: Change `'temp-' . $tempindex` for `"hpicfXcvrTemp.$tempindex"`

File: includes/discovery/sensors/voltage/procurve.inc.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        $limit_low                 = $entry['hpicfXcvrVccLoAlarm'] / $divisor_alarm;
+        $warn_limit_low            = $entry['hpicfXcvrVccLoWarn'] / $divisor_alarm;
+        $limit                     = $entry['hpicfXcvrVccHiAlarm'] / $divisor_alarm;
+        $warn_limit                = $entry['hpicfXcvrVccHiWarn'] / $divisor_alarm;
+        $current                   = $entry['hpicfXcvrVoltage'] / $divisor;
+        $entPhysicalIndex          = $index;
+        $entPhysicalIndex_measured = 'ports';
+        foreach ($dbquery as $dbindex => $dbresult) {
+            $descr = makeshortif($dbresult['ifDescr']) . ' Port Supply Voltage';
+            discover_sensor($valid['sensor'], 'voltage', $device, $oid, 'volt-' . $index, 'procurve', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
```
Comment (laf) [inline includes/discovery/sensors/voltage/procurve.inc.php]: Change `'volt-' . $index` for `"hpicfXcvrVoltage.$index"`

**Comment by angryp** [comment]
That was painful, but I think the changes have been pushed in the end.

As per the tests, I followed "Additional module support or test data" part. Snmprec file has been populated, but the following steps were faulty - snmpsim threw out a ton of errors on "save-test-data" step and "pre-commit" step succeeded at first run, but then started to fail. 
Pastebin output: https://pastebin.com/ddEzvBqL

**Comment by laf** [comment]
I've added the json test data on this, should be good once tests passes.

**Pull request merged**